README.md

```diff
@@ -1,2 +1,37 @@
-# cargo-template-cli
-Cargo template for generating CLIs
+# AWS Sigature Proxy
+
+A proxy whose entire job is to sign requests destined for the AWS API.
```

Comment:
I know this is a first PR, but would be nice to have a couple more sentences (even copy/pasted from your blog) about why this would be desirable to use and what this is trying to achieve.

Reply:
Yeah, good point!  I can probably summarize something here.

First attempt:

> # Motivation
>
> The AWS API requires a [non-standard signing process](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html) invented by Amazon, which makes it incompatible with many standard tools, as you can see from the need for projects like [`awscurl`](https://github.com/okigan/awscurl), also known as "curl but it works with AWS".
> 
> This project aims to fix that by hiding the non-standard signing process behind a proxy, so that any clients can use standard tools like `curl` (example below) and [OpenAPI](https://github.com/OpenAPITools/openapi-generator/) with standard clients.
> 
> That will hopefully allow us to take advantage of powerful open source tools that would otherwise be difficult to use with this API.  If you are bored and want to know why the proxy seemed like a good idea over other alternatives, feel free to read [this post](https://shaunverch.com/butter/open-source/2019/09/27/butter-days-6.html).
>
> # Usage
> ... etc. ...